### PR TITLE
VETTING:  enable reordering of letter upon letter expiry 

### PR DESCRIPTION
**Pull request by eunjuhuss**
#### Description:
when letter code has expired, incorrect Information was displayed in button and pop up incorrect modal.

#### Summary:
- Fixed an incorrect spelling for letter_expired props to display correct information when letter has expired.
- Added a function for colseModal to reuse for closing NotificaitonModal and ConfirmationModal

- To show notification modal for ordering letter 
```
const showNotificaitonModal = 
      (this.state.letter_sent_days_ago === undefined && !this.state.verifyingLetter) ||
      (this.state.letter_expired && this.state.letter_sent_days_ago >= 15);
```
- To show confirmation modal for confirming letter
 ```
   const showConfirmationModal = 
      this.state.verifyingLetter && this.state.letter_sent_days_ago < 15 && 
      !this.state.letter_expired && this.state.letter_sent !== "";
```

ps. For testing I have written an instruction for modifying letter sent data in eduid documentation (QUERY: UPDATE LETTER PROOFING SENT_DATA(IN LETTER DB) FOR SPECIFIC USER)

---
- When letter has expired
Screenshot 2020-11-05 at 15 09 52

- Clicking vetting button
Screenshot 2020-11-05 at 15 10 05

- Accept to send new letter
Screenshot 2020-11-05 at 15 10 15

---
#### For reviewer:
- [x] Is it displaying the correct text when letter has expired
- [x] When clicking vetting button, popping up notificaitonModal?
- [x] Accepting for ordering of new letter, displaying correct text in the vetting button?
- [x] Read the above description
- [x] Reviewed the code changes
- [x] Navigate to the branch (pulled the latest version)
- [x] Run the code and been able to execute the expected function
- [x] Check any styling on both desktop and mobile sizes

### CODE REVIEW

**Comment by eunjuhuss** [comment]
- Removed **console.logs** and **commented out** code
- Removed **Nin argument** to avoid sending id number to backend
- Removed "**letter_sent_dasy_ago**" in "showNotificationModal" and "showConfirmationModal"
- Separated functions for **close modal** 
```
   closeConfirmationModal=()=>{
    this.setState({
      verifyingLetter: false
    })
  }
```

```
 closeNotificationModal=()=>{
    this.setState({
      letter_sent: "",
      confirmingLetter: false,
      letter_expires: ""
    })
  }
```

File: src/components/LetterProofing.js [hunk 1]
```diff
@@ -7,39 +7,61 @@ import { shortCodePattern } from "../login/app_utils/validation/regexPatterns";
 class LetterProofingButton extends Component {
   state = {
     letter_expired: false,
-    letter_expires: false,
+    letter_expires: "",
     verifyingLetter: false,
     confirmingLetter: false,
-    letter_sent: "",
```
Comment (njons) [inline src/components/LetterProofing.js:13]: In the sprit of @fredrikt comments about naming, I wonder if these props could be a little more descriptive of what they pass on (i.e. that two are dates). I have also always wondered what the difference between the "confirming" prop and the "verifying" prop is here, maybe renaming could shed some light on it?

**Review by njons — COMMENTED** [review]
Pointed to some areas where naming could be considered (based on previous comments from @fredrikt on improving naming).

- button informs user that code has expired
- clicking button opens the a to "ACCEPT" letter sent to your address
- clicking "ACCEPT" updates the info in the button to a later date

**Comment by eunjuhuss** [comment]
- Installed jest for tests debugging and created launch.json file for configuration. debugging tested by running on the current test file. 
- Added unit tests for checking button information text in the vetting button, when id number is not added, letter has been sent and letter expired. 
For testing unit tests 
**npm run test-headless** /  to confirm test, adjust incorrect props in LetterProofing component and then run test, test should fail.

- Renamed props/states 
  - verifyngLetter -> verifyingLetter_sent
  - letter_sent -> letter_sent_date
  - letter_expired -> letter_expired_date

**Comment by eunjuhuss** [comment]
updated: 
- Replace slice() with Date()
- Reduce test string to prevent failure of test due to button text changes.
- New test for checking the child of description text (eg. The letter sent  yyyy-mm-dd) are including "Nan" and if empty string it should fail the tesl.

**Pull request merged**